phoenix-spark-base/src/main/java/org/apache/phoenix/spark/datasource/v2/reader/PhoenixInputPartitionReader.java

```diff
@@ -81,7 +86,7 @@ Properties getOverriddenPropsFromOptions() {
         return options.getOverriddenProps();
     }
 
-    private QueryPlan getQueryPlan() throws SQLException {
+    private QueryPlan getQueryPlan() throws SQLException, InvalidProtocolBufferException {
```

Comment:
I think you could catch the `InvalidProtocolBufferException` and throw it as a RuntimeException in `getQueryPlan()`, rather than doing it in `initialize()`

Reply:
done.